### PR TITLE
Add logs when Docker enabled selinux (closes #11197)

### LINKED ISSUE #11197: Proposal: Add some feedback info when user run daemon with --selinux-enabled option
The docs describes that "SELinux does not presently support the BTRFS storage driver". 
When i run Docker Daemon with command as follow, Docker Daemon is running successfuly,
    docker -d -s btrfs --selinux-enabled=true

I found the reason is that i have not config selinux, and function selinuxEnabled in sourcecode return false, so the error as follow was not implemented, the Docker Daemon goes on running.
    if selinuxEnabled() && config.EnableSelinuxSupport && driver.String() == "btrfs" {
		return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver!")
    }

As above, in my opinion, maybe we can add some logs about selinuxEnabled return result to advoid  user misunderstand it.

Thanks.

[Docker info]
Containers: 0
Images: 108
Storage Driver: btrfs
Execution Driver: native-0.2
Kernel Version: 3.13.0-24-generic
Operating System: Ubuntu 14.04 LTS
CPUs: 4
Total Memory: 3.666 GiB
Name: mabin-PC
ID: JRLZ:LAQV:CT2Z:WLEF:CCEB:J24M:HY75:GBOK:Y3A5:6Y2B:OLM3:CKXZ
No Proxy: localhost,127.0.0.0/8,::1

### PR DESCRIPTION
**Pull request by MabinGo**
The docs describes that "SELinux does not presently support the BTRFS storage driver". 
When i run Docker Daemon with command as follow, Docker Daemon is running successfuly,
docker -d -s btrfs --selinux-enabled=true

I found the reason is that i have not config selinux, and function selinuxEnabled in sourcecode return false, so the error as follow was not implemented, Docker Daemon goes on running.
if selinuxEnabled() && config.EnableSelinuxSupport && driver.String() == "btrfs" {
return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver!")
}

As above, in my opinion, maybe we can add some logs about selinuxEnabled return result to advoid user misunderstand it.

### CODE REVIEW

**Comment by crosbymichael** [comment]
@rhatdan can you take a look at this please?

**Comment by tiborvass** [comment]
Ping @MabinGo please address issues mentioned by rhatdan, thanks!

**Comment by MabinGo** [comment]
@rhatdan @tiborvass Thanks for your mention.I'm sorry to reply so late, because it's in the night in China 8 hours ago. I have checked the code with the mention by rhatdan, and found my carelessness, so i have updated the code again, please take a recheck, thanks all :-)

**Comment by MabinGo** [comment]
@rhatdan OK,i accept to. I have updated it, please take a recheck,thanks:-)

**Comment by MabinGo** [comment]
ping @crosbymichael @tiborvass @rhatdan Can you help me to check it please?Thanks

File: daemon/daemon.go [hunk 1]
```diff
@@ -880,9 +877,18 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine) (*Daemon, error)
...
-		return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver!")
+	if config.EnableSelinuxSupport {
+		if selinuxEnabled() {
+			// As Docker on btrfs and SELinux are incompatible at present, error on both being enabled
+			if driver.String() == "btrfs" {
+				return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver")
+			}
+			log.Debugf("SELinux enabled successfully")
+		} else {
+			log.Warnf("SELinux is not enabled on the host system. Docker is not able to enable SELinux")
```
Comment (tiborvass) [inline daemon/daemon.go]: @MabinGo `Docker could not enable SELinux`

Comment (MabinGo) [inline daemon/daemon.go]: Oh,my fault:(  I will update it right now.

**Comment by thaJeztah** [comment]
@MabinGo I think this PR is meant to resolve https://github.com/docker/docker/issues/11197?

If correct, could you add `closes #11197` to the description of the PR, so that GitHub automatically closes that issue if this is merged?

**Comment by MabinGo** [comment]
@thaJeztah Yes,this PR resolved to #11197,i'll add the tag just like you said.Thanks:-)
Should i add somthing as below?
this PR closes #11197

**Comment by MabinGo** [comment]
ping @thaJeztah @rhatdan @tiborvass Is that correct now?Thanks!

**Comment by thaJeztah** [comment]
@MabinGo correct, if you just add that to the description at the top of this page, it should be fine (no need to "ping" after that :smile:)

**Comment by MabinGo** [comment]
@thaJeztah Done.Thanks for your help, you are so nice:)

**Comment by rhatdan** [comment]
LGTM

File: daemon/daemon.go [hunk 2]
```diff
@@ -880,9 +877,18 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine) (*Daemon, error)
...
-		return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver!")
+	if config.EnableSelinuxSupport {
+		if selinuxEnabled() {
+			// As Docker on btrfs and SELinux are incompatible at present, error on both being enabled
+			if driver.String() == "btrfs" {
+				return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver")
+			}
+			log.Debugf("SELinux enabled successfully")
+		} else {
+			log.Warnf("SELinux could not enable on the host system. Docker is not able to enable SELinux")
```
Comment (tiborvass) [inline daemon/daemon.go]: @MabinGo this is still wrong. It should be `log.Warn("Docker could not enable SELinux on the host system")`.

File: daemon/daemon.go [hunk 3]
```diff
@@ -880,9 +877,18 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine) (*Daemon, error)
...
-	// As Docker on btrfs and SELinux are incompatible at present, error on both being enabled
-	if selinuxEnabled() && config.EnableSelinuxSupport && driver.String() == "btrfs" {
-		return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver!")
+	if config.EnableSelinuxSupport {
+		if selinuxEnabled() {
+			// As Docker on btrfs and SELinux are incompatible at present, error on both being enabled
+			if driver.String() == "btrfs" {
+				return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver")
+			}
+			log.Debugf("SELinux enabled successfully")
```
Comment (tiborvass) [inline daemon/daemon.go]: `log.Debug(`

Comment (MabinGo) [inline daemon/daemon.go]: Why could not use log.Debugf?

File: daemon/daemon.go (hunk 2 above)
Comment (MabinGo) [inline daemon/daemon.go]: current log info is followed to rhatdan's suggestion.
"SELinux could not enable on the host system" means that Selinux did not enable on the host system, "Docker is not able to enable SELinux" means that Docker enabled Selinux failure.In my opinion,maybe we could keep current situation,what do you think?

File: daemon/daemon.go (hunk 3 above)
Comment (thaJeztah) [inline daemon/daemon.go]: Because you're not using a format-string (e.g. `'Some %s'`), Debug() is sufficient, thus cleaner code.

Comment (MabinGo) [inline daemon/daemon.go]: Ok,i see,i have checked the functions of Debug and Debugf, then understanded difference.
In addition, i have checked the whole daemon/daemon.go, and found the same situation of mine, just like "log.Debugf("Creating images graph")" and so on, so maybe we can use "log.Debug" to replace "log.Debugf", what's your opinion?Thanks:-)

Comment (MabinGo) [inline daemon/daemon.go]: updated the code which i newly added to.

**Comment by MabinGo** [comment]
@tiborvass @rhatdan @thaJeztah Is the modifies of this PR right now?

File: daemon/daemon.go (hunk 3 above)
Comment (thaJeztah) [inline daemon/daemon.go]: > so maybe we can use "log.Debug" to replace "log.Debugf", what's your opinion?

Not in *this* PR, but you could do that in a separate PR (to keep things separated)

**Comment by thaJeztah** [comment]
@MabinGo I think this is how @tiborvass wanted it yes. But he should review it, because I'm not a very good "Go" coder myself (yet) :)

**Comment by MabinGo** [comment]
@thaJeztah you are modest :-) I'm a Go language and opensource beginner, then I'll learn from everybody.

File: daemon/daemon.go [hunk 4]
```diff
@@ -880,9 +877,18 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine) (*Daemon, error)
...
-		return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver!")
+	if config.EnableSelinuxSupport {
+		if selinuxEnabled() {
+			// As Docker on btrfs and SELinux are incompatible at present, error on both being enabled
+			if driver.String() == "btrfs" {
+				return nil, fmt.Errorf("SELinux is not supported with the BTRFS graph driver")
+			}
+			log.Debug("SELinux enabled successfully")
+		} else {
+			log.Warn("SELinux could not enable on the host system. Docker is not able to enable SELinux")
```
Comment (tiborvass) [inline daemon/daemon.go]: This sentence is incorrect. I'll repeat what I said earlier: `Docker could not enable SELinux`. Ping @SvenDowideit @moxiegirl @fredlf

Comment (fredlf) [inline daemon/daemon.go]: I agree with @tiborvass . His revision is clearer and grammatically correct. Cheers.

Comment (MabinGo) [inline daemon/daemon.go]: @tiborvass @fredlf I get it. Thanks for your advice:-) I have updated the code.

**Comment by tiborvass** [comment]
LGTM
